src-opam/dockerfile_opam.ml

```diff
@@ -81,23 +81,32 @@ let bubblewrap_and_dev_packages distro =
     @@ dev_packages ()
 
 let install_bubblewrap_wrappers =
-  (* Enable bubblewrap *)
-  run "echo 'wrap-build-commands: []' > ~/.opamrc-nosandbox" @@
-  run "echo 'wrap-install-commands: []' >> ~/.opamrc-nosandbox" @@
-  run "echo 'wrap-remove-commands: []' >> ~/.opamrc-nosandbox" @@
-  run "echo 'required-tools: []' >> ~/.opamrc-nosandbox" @@
-  run "echo '#!/bin/sh' > /home/opam/opam-sandbox-disable" @@
-  run "echo 'cp ~/.opamrc-nosandbox ~/.opamrc' >> /home/opam/opam-sandbox-disable" @@
-  run "echo 'echo --- opam sandboxing disabled' >> /home/opam/opam-sandbox-disable" @@
+  let strip = true in
+  let opamrc_sandbox = heredoc ~strip
+{|	wrap-build-commands: ["%%{hooks}%%/sandbox.sh" "build"]
+	wrap-install-commands: ["%%{hooks}%%/sandbox.sh" "install"]
+	wrap-remove-commands: ["%%{hooks}%%/sandbox.sh" "remove"]|} in
```

Comment:
I would add the `|} in` on a newline wherever you use it. Makes it easier to read and show what is multiline string and what is the actual command.

Reply:
The problem is that the end delimiter has to go on a newline. The formatting function will add `"\n%s" t.delimiter` (usually `\nEOF`), so adding a new line here would render 

```
COPY --chown=opam <<-EOF /home/opam/opam-sandbox-disable
	#!/bin/sh
	cp ~/.opamrc-nosandbox ~/.opamrc
	echo --- opam sandboxing disabled

EOF
```

instead of
```
COPY --chown=opam <<-EOF /home/opam/opam-sandbox-disable
	#!/bin/sh
	cp ~/.opamrc-nosandbox ~/.opamrc
	echo --- opam sandboxing disabled
EOF
```

I'm not sure checking whether the heredoc ends with a newline or not is good, because the heredoc should be preserved verbatim. What to you think?

Reply:
The more compact version makes sense.